proxy_presentation.php

```diff
@@ -0,0 +1,96 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Presentation proxy script.
+ *
+ * @package   mod_bigbluebuttonbn
+ * @copyright 2010 onwards, Blindside Networks Inc
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author    Kenneth Hendricks (kennethhendricks@catalyst-au.net)
+ */
+
+require_once(dirname(__FILE__) . '/../../config.php');
+require_once($CFG->libdir . '/filelib.php');
+require_once(dirname(__FILE__) . '/locallib.php');
+
+$relativepath = get_file_argument();
+
+if (empty($relativepath)) {
+    exit();
+}
+
+$pathparts = explode('/', $relativepath);
+
+// If path is of format /presentation/{recording-id}/xxxx we need to check
+// user has access to the recording in question. Urls of this format contain
+// sensitive media.
+if (isset($pathparts[1]) && $pathparts[1] === 'presentation' && isset($pathparts[2])) {
+    $recordingid = $pathparts[2];
+    $recordings = bigbluebuttonbn_get_recordings_array_fetch_page([], [$recordingid]);
+
+    if (!isset($recordings[$recordingid])) {
+        throw new moodle_exception("invalidaccessparameter");
+    }
+
+    $meetingid = $recordings[$recordingid]['meetingID'];
+
+    $meetingidparts = explode('-', $meetingid);
+    if (!isset($meetingidparts[2])) {
+        throw new moodle_exception("invalidaccessparameter");
+    }
+
+    $bbbinstanceid = $meetingidparts[2];
+    $viewinstance = bigbluebuttonbn_view_instance_bigbluebuttonbn($bbbinstanceid);
+    require_login($viewinstance['course'], true, $viewinstance['cm']);
+} else {
+    require_login();
+}
+
+// One JS asset contains a non relative route we need to rewrite
+// to be relative in order to work with the proxy. We also filter out the
+// Content-Length Header in this case to avoid truncation
+$jstobereplaced = false;
```

Comment:
Could this be shortened to a ternary? Not sure how messy it would be but 4 lines -> 1 line

Reply:
yea agreed, we can actually just evaluate the condition here instead